upload_artifacts.sh

```diff
@@ -1,21 +1,33 @@
 #!/bin/bash
 
-# Used in .travis.yml
+# This script is used in .travis.yml
+
+set -e
 
 BUCKET=piksi-buildroot-images
 PIKSI_VERSION=v3
+BUILD_VERSION=$(git rev-parse --short HEAD)
 
-set -e
+if [ "$TRAVIS_PULL_REQUEST" == "true" ]; then
+  FOLDER=pull_requests
+else
+  if [ "$TRAVIS_BRANCH" == "master" ]; then
+    FOLDER=master
+  else
+    FOLDER=misc
+  fi
+fi
 
-BUILD_VERSION=$(git rev-parse --short HEAD)
-BUILD_DIR=commit_$BUILD_VERSION
-# "folders" on S3 are prefix keys
-# http://stackoverflow.com/questions/33113354/how-to-create-a-folder-like-i-e-prefix-object-on-s3-using-the-aws-cli
-aws s3api put-object --bucket $BUCKET --key $PIKSI_VERSION/$BUILD_DIR/
-UPLOAD_DIR="s3://$BUCKET/$PIKSI_VERSION/$BUILD_DIR"
+BUILD_DIR="UTC-$(date -u +%T)_TRAVIS-$(echo $TRAVIS_BUILD_NUMBER)_COMMIT-$(echo $BUILD_VERSION)"
```

Comment:
Should use iso8601 for the date - don't need the prefixs for travs or commit - that should be self-evident:

```
BUILD_DIR="$(date -u +"%Y-%m-%dT%H:%M:%SZ")-$TRAVIS_BUILD_NUMBER-$BUILD_VERSION"
```